### PR TITLE
Support SDS011 component.

### PR DESCRIPTION
**Pull request by puuu**
## Description:

**Related issue (if applicable):**

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#185
**Pull request in [esphome-core](https://github.com/esphome/esphome-core) with C++ framework changes (if applicable):** esphome/esphome-core#538

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphomedocs](https://github.com/OttoWinter/esphomedocs).

### CODE REVIEW

File: esphome/components/sensor/sds011.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+from esphome.cpp_types import App, Component
+
+DEPENDENCIES = ['uart']
+
+SDS011Component = sensor.sensor_ns.class_('SDS011Component', uart.UARTDevice, Component)
+SDS011Sensor = sensor.sensor_ns.class_('SDS011Sensor', sensor.EmptySensor)
+
+
+def validate_sds011_rx_mode(value):
+    if (CONF_QUERY_MODE in value and CONF_RX_ONLY in value
```
[the remark this answers is not in the document]
Comment (puuu) [reply esphome/components/sensor/sds011.py]: Yes, this looks better, changed.

File: esphome/components/sensor/sds011.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            # Check for TimePeriod instance ('never' update interval)
+            if (update_interval.milliseconds or 0) != 0 or (update_interval.seconds or 0) != 0:
+                # Check if time period is multiple of minutes
+                raise vol.Invalid("Maximum update interval precision in non-rx_only mode is 1min")
+            if update_interval.total_minutes > 30:
+                raise vol.Invalid("Maximum update interval is 30min")
+    return value
+
+
+PMSX003_SENSOR_SCHEMA = sensor.SENSOR_SCHEMA.extend({
```
Comment (OttoWinter) [inline esphome/components/sensor/sds011.py]:
```suggestion
SDS011_SENSOR_SCHEMA = sensor.SENSOR_SCHEMA.extend({
```

File: esphome/components/sensor/sds011.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+})
+
+PLATFORM_SCHEMA = vol.All(sensor.PLATFORM_SCHEMA.extend({
+    cv.GenerateID(): cv.declare_variable_id(SDS011Component),
+    cv.GenerateID(CONF_UART_ID): cv.use_variable_id(UARTComponent),
+
+    vol.Optional(CONF_QUERY_MODE): cv.boolean,
+    vol.Optional(CONF_RX_ONLY): cv.boolean,
+
+    vol.Optional(CONF_PM_2_5): cv.nameable(PMSX003_SENSOR_SCHEMA),
```
Comment (OttoWinter) [inline esphome/components/sensor/sds011.py]:
```suggestion
    vol.Optional(CONF_PM_2_5): cv.nameable(SDS011_SENSOR_SCHEMA),
```

File: esphome/components/sensor/sds011.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    vol.Optional(CONF_PM_10_0): cv.nameable(PMSX003_SENSOR_SCHEMA),
```
Comment (OttoWinter) [inline esphome/components/sensor/sds011.py]:
```suggestion
    vol.Optional(CONF_PM_10_0): cv.nameable(SDS011_SENSOR_SCHEMA),
```

File: esphome/components/sensor/sds011.py (hunk 2 above)
Comment (puuu) [reply esphome/components/sensor/sds011.py]: Good point. Shame on me. :-(

**Review by OttoWinter — APPROVED** [review]

**Comment by puuu** [comment]
The last commits removed the query mode. The last open question is about `update_interval`, since this name comes from the PollingComponent, but we do not inherit from PollingComponent. PollingComponent call `set_interval()` and we should avoid this for `update_interval = 0`.

**Pull request merged**